### PR TITLE
Added Alternative optional replacement for float added

### PR DESCRIPTION
**Pull request by BlazesRus**
Requires 5 bytes instead of 4 bytes of float but stores precisely at 4 decimals places (and +- uint 16 stored in whole half).  Use AlternativeCode build to switch to replacement type usage. (Or just make new config settup with PoESkillTree_UseSmallDec_ForAttributes preprocessor flagged)

### CODE REVIEW

**Comment by BlazesRus** [comment]
AlternativeTypes configs need fixed currently for display of the numbers on the bars still but normal non-AlternativeType builds have been tested by me to work right(Stress tested the x64 build with 300 build generations).

**Comment by BlazesRus** [comment]
Compiling at x64 fixes the out of memory crash from generating too many builds with usage of lots pseudo attributes

**Comment by MLanghof** [comment]
...could you elaborate on why you think this is needed?

**Comment by brather1ng** [comment]
Yeah, like @MLanghof, I don't see the use in this. Floats can accurately store values with 6 significant digits, so why would we want to use something that is bigger and can store less digits?

**Comment by BlazesRus** [comment]
Fixed the build now

**Comment by BlazesRus** [comment]
More accurate display of the build differences

**Comment by BlazesRus** [comment]
float does not store every decimal from 1-6 digits

**Comment by BlazesRus** [comment]
Fixes those sometimes 0.1 to 0.2 difference in displayed attribute differences

**Comment by BlazesRus** [comment]
I could add option preprocessor flag to reduce SmallDec down by 1 byte in size while still storing same amount of possible represented values and 100% accuracy stored values at small extra cost of processing time (since plenty of unused ushort value in DecimalStatus)

**Comment by BlazesRus** [comment]
https://github.com/BlazesRus/PoESkillTree/releases/tag/ReleaseCompile_3_26_2017_X64
is compiled build (64 bit build)

**Comment by EmmittJ** [comment]
A float has a precision, in C#, of 7 digits. If you need more then you can
use a double (16 digits) or decimal (28 digits). I am somewhat confused as
to what this implementation brings. You are just always storing up to 4
decimal places?

**Comment by MLanghof** [comment]
I guess he wants to avoid the 0.99999997 kind of thing. But the solution to that is to round appropriately, not to replace floats.

File: WPFSKillTree/GlobalCode_ExperimentalCode/SuperDec_SmallDec_TypeConverter.cs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+			if (destinationType == typeof(int))
+			{
+				return (int)value;
+			}
+			return base.ConvertTo(context, culture, value, destinationType);
+		}
+	}
+}
+
+//Some Converter code from ArithmeticConverter.cs in PoESkillTree code (required license for related part of code below)
```
Comment (MLanghof) [inline WPFSKillTree/GlobalCode_ExperimentalCode/SuperDec_SmallDec_TypeConverter.cs]: ...why are you putting license references to our code into our code?

File: WPFSKillTree/ViewModels/Attribute.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ public Attribute(string text)
             Text = text;
         }
 
-        public override string ToString()
+		public override string ToString()
```
Comment (MLanghof) [inline WPFSKillTree/ViewModels/Attribute.cs]: https://imgur.com/a/lF4AP

File: WPFSKillTree/GlobalCode_ExperimentalCode/SuperDec_SmallDec_TypeConverter.cs (hunk 1 above)
Comment (BlazesRus) [inline WPFSKillTree/GlobalCode_ExperimentalCode/SuperDec_SmallDec_TypeConverter.cs]: Removed

File: WPFSKillTree/ViewModels/Attribute.cs (hunk 2 above)
Comment (BlazesRus) [inline WPFSKillTree/ViewModels/Attribute.cs]: Replaced those with tabs with (4x Spaces)

**Comment by BlazesRus** [comment]
Optional flag based config that replaces float with SmallDec for attributes values (not things like weight and position) for  a fix for the:
     * - Mana regeneration shows sometimes incorrect value (0.1 difference from in-game value).
     * - Damage type ranges shows sometimes incorrect value affecting overall DPS (1 difference from in-game value, occurs with damage conversions).
     * - Damage per Second shows sometimes incorrect value (0.1 difference from in-game value).
     * - Spell Critical Strike chance shows sometimes incorrect value (0.1 difference from in-game value).
     * - Estimated chance to Evade Attacks shows sometimes incorrect value.
listed on compute.cs (and fix for sometimes inaccurate display of hp regeneration percent display comparisons sometimes being off (sometimes displaying difference of 1 value when not really full value of difference)

**Pull request closed without merging**